lib/Reader/readerir.cpp

```diff
@@ -1678,6 +1762,44 @@ Type *GenIR::getClassTypeWorker(
     }
   }
 
+  uint32_t EEClassSize = 0;
+  bool HaveClassSize = false;
+  if (!IsRefClass) {
+    try {
+      EEClassSize = getClassSize(ClassHandle);
+      HaveClassSize = true;
+    } catch (...) {
```

Comment:
Is there any way to query the class handle up front to see if asking the runtime for size would cause an exception?